[T7][W13-B4]

R-o-y opened the pull request:
update the following documentations:
* user guide
* developer guide

K1ang commented on docs/UserGuide.md:
```diff
@@ -1,156 +1,224 @@
+   * **`delete 3`** : deletes the 3rd task shown in the current list
    * **`exit`** : exits the app
 6. Refer to the [Features](#features) section below for details of each command.<br>
 
```
How about including a mock up of the UI to show the users here?

K1ang commented on docs/UserGuide.md:
```diff
@@ -1,156 +1,224 @@
-
-### 2.3. Listing all persons : `list`
+* `do group meeting \from 4pm today \to 6pm today \remind_at 3pm \as important \under school_work CS2103T`
+* `add submit pre-tutorial activities \by 6pm this Wednesday \remind_at 5pm this Wednesday \every week`
```
- How about covering the basic date format for example instead of natural language kind of date (today, tomorrow, wednesday) first?
- This is a little complicated here. Try not to have multiple command names first. Or allow aliases for command instead.

K1ang commented on docs/UserGuide.md:
```diff
@@ -1,156 +1,224 @@
+  Suppose originally, the description of this task is `watch nba`,
+  it will be change to `watch NBA`.
+  The reminder time will also be updated to the specified one: `this Sunday 8am`.
+
```
As your parameters are optional, more thinking can be placed for this `edit` function.
Maybe you can explain more, like what will happen to the existing values that are not specified in a particular `edit` command.
Eg. edit 2 \desc Assignment
- What will happen to the dates that were earlier on but not included now?
- Or what if I intend to leave out a start time to convert it from a task that only have deadline, etc.

K1ang commented on docs/UserGuide.md:
```diff
@@ -1,156 +1,224 @@
-Format: `clear`
+Undo the command previous to the current command.<br>
+Format: `undo`  
 
```
More thoughts can be given to the undo function too, such as what type of commands can be undo-ed.

K1ang reviewed (commented):
Good job! Keep it up!

K1ang commented:
The User Guide has been reviewed. The DevGuide will be reviewed soon, so do not close this PR yet.

leeyh20 committed:
Edit Config to use absolute path instead

R-o-y committed:
add tests for highlight feature

R-o-y committed:
add test cases using pressing key or key combinations
TODO: refactoring test cases

R-o-y committed:
refactoring undo redo test

R-o-y committed:
fix style issue add "event.consume"

R-o-y committed:
add content assistance for parameter keys and add corresponding test cases

R-o-y committed:
Merge pull request #87 from CS2103JAN2017-W13-B4/auto-completion

Auto completion feature

leeyh20 committed:
Save at command, file will be moved when change directory. Or at least it tries.

R-o-y committed:
implement remove_alias command

leeyh20 committed:
Fix tests

R-o-y committed:
fix codacy issue

R-o-y committed:
Merge pull request #88 from CS2103JAN2017-W13-B4/remove_alias

implement remove_alias command

ramjanarthan committed:
Add new TaskDate class

ramjanarthan committed:
Add new class to store times and date

ramjanarthan committed:
Change code to use new class

ramjanarthan committed:
Change parser to use TaskDate class

ramjanarthan committed:
Add toString method

ramjanarthan committed:
Change class to use new DateTime class

ramjanarthan committed:
Change class to use new TaskDate

ramjanarthan committed:
Change code to use TaskDate class

ramjanarthan committed:
Make test cases pass

ramjanarthan committed:
Shift Parsing date to this class

ramjanarthan committed:
Refactor code to use TaskDate.parseDate

leeyh20 committed:
Fix codacy

leeyh20 committed:
Merge branch 'master' into time

# Conflicts:
#	src/main/java/seedu/doist/model/task/ReadOnlyTask.java
#	src/main/java/seedu/doist/model/task/Task.java
#	src/main/java/seedu/doist/storage/StorageManager.java
#	src/main/java/seedu/doist/ui/PersonCard.java

leeyh20 committed:
Fix date on task card

leeyh20 committed:
Fix messages

leeyh20 committed:
Remove PersonCard since it has been renamed to TaskCard

leeyh20 committed:
Change taskcard colour when overdue task

ramjanarthan committed:
Use libraries to support near-match for search

leeyh20 committed:
Change messages of list, default view is to see NOT FINISHED tasks

leeyh20 committed:
Fix tests. Btw LogicManagerTest is testing both model and logic without UI

ramjanarthan committed:
Create method to get all Task names

ramjanarthan committed:
Add auto-complete feature for search

ramjanarthan committed:
Allow MainWindow to use Model

leeyh20 committed:
Add sort by multiple parameters, in different order, all up to you

leeyh20 committed:
Save file after sorting

leeyh20 committed:
Default sorting when load

leeyh20 committed:
If testing, don't sort tasks...

leeyh20 committed:
Merge pull request #91 from CS2103JAN2017-W13-B4/time

Time

R-o-y committed:
change icon

R-o-y committed:
add parsing range

R-o-y committed:
fix coday issue (also a potential bug)

leeyh20 committed:
change icon

leeyh20 committed:
Rename icon

ramjanarthan committed:
Remove dependency on Model in Ui

ramjanarthan committed:
Modify the TAB button for auto-complete for Find command

leeyh20 committed:
Change dark theme

leeyh20 committed:
Change test image name

leeyh20 committed:
Merge pull request #95 from CS2103JAN2017-W13-B4/change_icon

change icon and dark theme

leeyh20 committed:
Merge branch 'master' into change-storage-path

# Conflicts:
#	src/main/java/seedu/doist/logic/commands/Command.java
#	src/main/java/seedu/doist/logic/commands/SortCommand.java
#	src/main/java/seedu/doist/model/AliasListMap.java
#	src/main/java/seedu/doist/model/Model.java
#	src/main/java/seedu/doist/model/ModelManager.java
#	src/main/java/seedu/doist/storage/StorageManager.java
#	src/test/java/guitests/AliasCommandTest.java
#	src/test/java/guitests/ResetAliasCommandTest.java
#	src/test/java/guitests/SortCommandTest.java

leeyh20 committed:
fix tests

leeyh20 committed:
Fix save as command to work with "../"

leeyh20 committed:
Some fixes

leeyh20 committed:
Update UI when path changes

ramjanarthan committed:
Refactor code in EditParser

leeyh20 committed:
Fix bug when preferences is deleted

ramjanarthan committed:
Improve code quality and pass tests

leeyh20 committed:
add SaveAsLogicTest

leeyh20 committed:
Upload correct test file

ramjanarthan committed:
Merge branch 'master' into search

ramjanarthan committed:
Merge pull request #92 from CS2103JAN2017-W13-B4/search

Search near-match

leeyh20 committed:
Try to fix test

leeyh20 committed:
Merge branch 'master' into change-storage-path

leeyh20 committed:
Fix import

leeyh20 committed:
Update AboutUs.md

ramjanarthan committed:
Update AboutUs-Ram

leeyh20 committed:
Update AboutUs.md

ramjanarthan committed:
Tag code

ramjanarthan committed:
Include collated files

leeyh20 committed:
Some fixes, codacy

leeyh20 committed:
Try to remove one test

leeyh20 committed:
Remove incorrect path test

leeyh20 committed:
Merge pull request #98 from CS2103JAN2017-W13-B4/change-storage-path

Change storage path

R-o-y committed:
Merge pull request #96 from CS2103JAN2017-W13-B4/hyphen

add parsing range

R-o-y committed:
Update AboutUs.md

R-o-y committed:
update collated

R-o-y committed:
Merge pull request #100 from CS2103JAN2017-W13-B4/update-about-us

Update AboutUs.md

leeyh20 committed:
Fix dark theme

ramjanarthan committed:
Fix bug in edit

ramjanarthan committed:
Add message for Invalid dates

ramjanarthan committed:
Use error message while throwing exceptions

ramjanarthan committed:
Use the error message for exception

ramjanarthan committed:
Add test cases for EditParser

ramjanarthan committed:
Merge pull request #101 from CS2103JAN2017-W13-B4/time

Refactor code in EditParser

leeyh20 committed:
Add comment

leeyh20 committed:
Change edit, find will jump to task. Show finished tasks at the bottom. Auto sort the tasks and fix tests

leeyh20 committed:
Merge pull request #112 from CS2103JAN2017-W13-B4/select-list

Jump to task. Show finished tasks. Auto Sort.

leeyh20 committed:
Trying to fix key combination feedback box

R-o-y committed:
feedback to user if undo and redo already reach the top or bottom

ramjanarthan committed:
Change date format in UI display

leeyh20 committed:
Don't update filtered list to show default

leeyh20 committed:
Merge branch 'fix_jumpto' into undo_message

ramjanarthan committed:
Define a compareTo function

leeyh20 committed:
fix tests

leeyh20 committed:
Fix equals method in Date to pass test

R-o-y committed:
fix the following 2 problems:
1. store and recover the last un-entered input (rather than use empty string)
2. when execute 2 consecutive commands, and use command navigation, the second one navigated to will be highlighted incorrectly

ramjanarthan committed:
Change the way suggestions work for 'Find'

leeyh20 committed:
Fix bug

leeyh20 committed:
Fix codacy

leeyh20 committed:
Merge pull request #113 from CS2103JAN2017-W13-B4/fix_jumpto

Fix jumpto

leeyh20 committed:
Fix tabs

ramjanarthan committed:
Remove all reminder and recurring tasks

leeyh20 committed:
Fix tests

leeyh20 committed:
Merge branch 'master' into undo_message

leeyh20 committed:
Merge pull request #114 from CS2103JAN2017-W13-B4/undo_message

Feedback to user if undo and redo already reach the top or bottom

leeyh20 committed:
Change UI image

ramjanarthan committed:
Tweak the parameters for Find command

ramjanarthan committed:
Reflect new design for Find

ramjanarthan committed:
Create new class to encapsulate all string matching heuristics

ramjanarthan committed:
Create custom compareTo function

ramjanarthan committed:
Add method to filter tasks by date

leeyh20 committed:
Merge branch 'master' into documentation-changes

leeyh20 committed:
Add command summary

leeyh20 committed:
Mention highlighting and autocomplete

R-o-y committed:
Merge pull request #116 from CS2103JAN2017-W13-B4/documentation-changes

Documentation changes

R-o-y committed:
add tests and refatoring

ramjanarthan committed:
Modify the algorithm to search using individual words

ramjanarthan committed:
Sort the tasks after a find, to show most relevant tasks first

ramjanarthan committed:
Modify the compareTo function

ramjanarthan committed:
Change the month offset

ramjanarthan committed:
Include capability for \from and \to search

leeyh20 committed:
Allow horizontal scrolling of tasks

ramjanarthan committed:
Change code to allow list time, task type and under tag

ramjanarthan committed:
Remove throwing exception for no time parameters

leeyh20 committed:
Fix tests

leeyh20 committed:
Fix scrollbars

leeyh20 committed:
Add stars to priority

leeyh20 committed:
Update ppt and change name of log

leeyh20 committed:
Update ppt and change name of log

leeyh20 committed:
Add two images for user guide

leeyh20 committed:
Add two images for user guide

leeyh20 committed:
Edits to user guide

leeyh20 committed:
Edits to user guide

R-o-y committed:
remove unused parameter key

R-o-y committed:
new collate

R-o-y committed:
new collate

R-o-y committed:
new collate

ramjanarthan committed:
Change autocomplete text substitution

leeyh20 committed:
new collate

R-o-y committed:
new collate

R-o-y committed:
fix typo

R-o-y committed:
Merge branch 'roy-collate' of https://github.com/CS2103JAN2017-W13-B4/main into roy-collate

ramjanarthan committed:
Fix codacy issues

R-o-y committed:
new test cases

R-o-y committed:
add more test cases for finish command

leeyh20 committed:
Manual testing test script and sample data

leeyh20 committed:
Manual testing test script and sample data

ramjanarthan committed:
Collate and refactor code

ramjanarthan committed:
Collate code

ramjanarthan committed:
Collate more code

leeyh20 committed:
Change star color

leeyh20 committed:
Change dark theme

leeyh20 committed:
Refactor Config to have its own ConfigManager so that ModelManager.java isnt so long

leeyh20 committed:
Refactor into AliasListMapModel to make ModelManager shorter.
Fix tests.

leeyh20 committed:
Remove all occurances of person and addressbook

leeyh20 committed:
Merge pull request #118 from CS2103JAN2017-W13-B4/roy-collate

Collate and test cases

leeyh20 committed:
Merge pull request #119 from CS2103JAN2017-W13-B4/load

Refactoring

leeyh20 committed:
Minor changes

leeyh20 committed:
Merge branch 'master' into load

leeyh20 committed:
Collate

leeyh20 committed:
Comments

leeyh20 committed:
add LoadCommand to load data

ramjanarthan committed:
Add tests for list \from \to

ramjanarthan committed:
Collate some tests

leeyh20 committed:
Add LoadCommandTest

leeyh20 committed:
Merge branch 'master' into enhancements

leeyh20 committed:
Revert History and fix build

leeyh20 committed:
Merge pull request #117 from CS2103JAN2017-W13-B4/enhancements

Enhancements

leeyh20 committed:
Merge pull request #120 from CS2103JAN2017-W13-B4/load

LoadCommand

leeyh20 committed:
Collate

leeyh20 committed:
Add help window

leeyh20 committed:
Add two tests

leeyh20 committed:
Add load command

leeyh20 committed:
Merge branch 'documentation-changes' of https://github.com/CS2103JAN2017-W13-B4/main into documentation-changes

# Conflicts:
#	docs/UserGuide.md

leeyh20 committed:
Update cheatsheet

leeyh20 committed:
More pictures

R-o-y committed:
add quick navigation feature

R-o-y committed:
udpate mode help message

ramjanarthan committed:
Add activity diagram ppt

ramjanarthan committed:
Add activity diagram png

ramjanarthan committed:
Add activity diagram

ramjanarthan committed:
Update ram's features

R-o-y committed:
add navigation and editing mode test

R-o-y committed:
udpate mode test

R-o-y committed:
add test cases for use J and K to navigate task list up and down

R-o-y committed:
update comment, rename manager to controller to conform to MVC pattern

R-o-y committed:
Update AboutUs.md

R-o-y committed:
Merge pull request #121 from CS2103JAN2017-W13-B4/quick-navigation

Quick navigation

leeyh20 committed:
Merge branch 'documentation-changes' of https://github.com/CS2103JAN2017-W13-B4/main into documentation-changes

leeyh20 committed:
Add some alias commands

leeyh20 committed:
Fix figure 3.2.2

leeyh20 committed:
About us

leeyh20 committed:
Update portions of developer guide

leeyh20 committed:
Minor change

leeyh20 committed:
Ui pictures

ramjanarthan committed:
Collate code

ramjanarthan committed:
Upload higher quality activity diagram image

leeyh20 committed:
ModelClassDiagram

leeyh20 committed:
Merge branch 'documentation-changes' of https://github.com/CS2103JAN2017-W13-B4/main into documentation-changes

R-o-y committed:
refactoring

leeyh20 committed:
Update developer guide

R-o-y committed:
Merge branch 'documentation-changes' of https://github.com/CS2103JAN2017-W13-B4/main into documentation-changes

leeyh20 committed:
Minor changes

leeyh20 committed:
Merge branch 'documentation-changes' of https://github.com/CS2103JAN2017-W13-B4/main into documentation-changes

leeyh20 committed:
Change user guide

leeyh20 committed:
Add more tests

leeyh20 committed:
Merge branch 'master' into documentation-changes

leeyh20 committed:
More tests

leeyh20 committed:
Merge pull request #122 from CS2103JAN2017-W13-B4/more-tests

Add more tests!

ramjanarthan committed:
Minor fixes, remove 'List in' feature

ramjanarthan committed:
Add some design decisions

ramjanarthan committed:
Add some commands

leeyh20 committed:
Collate

R-o-y committed:
check github blame history

leeyh20 committed:
Fix links

leeyh20 committed:
Minor changes

R-o-y committed:
check github blame history

leeyh20 committed:
Increase SLAP

R-o-y committed:
almost done with collating

R-o-y committed:
refactoring

ramjanarthan committed:
Collate some tests

ramjanarthan committed:
Merge branch 'master' of https://github.com/CS2103JAN2017-W13-B4/main

R-o-y committed:
refactoring and more test cases

R-o-y committed:
Merge branch 'master' into Roy-final-collated

ramjanarthan committed:
Add links to features

R-o-y committed:
Merge pull request #124 from CS2103JAN2017-W13-B4/Roy-final-collated

Roy final collated

R-o-y committed:
udpate collate

ramjanarthan committed:
Add list, find, delete and edit commands

ramjanarthan committed:
Include undo, redo, clear and list_tag commands

ramjanarthan committed:
Remove \\ in description

ramjanarthan committed:
Remove the word 'abstracted'

R-o-y committed:
update UI diagram

R-o-y committed:
Update DeveloperGuide.md

remove slash

R-o-y committed:
refactoring

R-o-y committed:
update collate

R-o-y committed:
fix checkstyle issue

leeyh20 committed:
user guide

R-o-y committed:
Update AboutUs.md

add link in about us

leeyh20 committed:
Merge branch 'documentation-changes' of https://github.com/CS2103JAN2017-W13-B4/main into documentation-changes

leeyh20 committed:
Collate and refactor

leeyh20 committed:
Refactor

leeyh20 committed:
TestScript and user guide

leeyh20 committed:
Merge pull request #123 from CS2103JAN2017-W13-B4/documentation-changes

Documentation changes

leeyh20 committed:
Delete Ui.png

leeyh20 committed:
Add UI.png

leeyh20 committed:
Update readme

leeyh20 committed:
Update about us

leeyh20 committed:
Fix about us links again

leeyh20 committed:
Update one line at collate
